services/gossip/adapter/tcp/outgoing_connection.go

```diff
@@ -52,20 +52,37 @@ func newOutgoingConnection(peer adapter.TransportPeer, parentLogger log.Logger,
 	queue.networkAddress = networkAddress
 	queue.Disable() // until connection is established
 
+	sendErrors, sendQueueErrors := generateMetrics(peerHexAddress, metricFactory)
+
 	client := &outgoingConnection{
 		logger:          logger,
 		sharedMetrics:   sharedMetrics,
 		metricRegistry:  metricFactory,
 		config:          transportConfig,
 		queue:           queue,
 		peerHexAddress:  peerHexAddress,
-		sendErrors:      metricFactory.NewGauge(fmt.Sprintf("Gossip.OutgoingConnection.SendError.%s.Count", peerHexAddress)),
-		sendQueueErrors: metricFactory.NewGauge(fmt.Sprintf("Gossip.OutgoingConnection.EnqueueErrors.%s.Count", peerHexAddress)),
+		sendErrors:      sendErrors,
+		sendQueueErrors: sendQueueErrors,
 	}
 
 	return client
 }
 
+// This is a round-about way to clean up these metrics that can be left over from previous connection
+func generateMetrics(peerHexAddress string, metricFactory metric.Registry) (*metric.Gauge, *metric.Gauge){
+	sendErrorsName := fmt.Sprintf("Gossip.OutgoingConnection.SendError.%s.Count", peerHexAddress)
+	sendErrorMetric := metricFactory.Get(sendErrorsName)
+	if sendErrorMetric != nil {
+		metricFactory.Remove(sendErrorMetric)
```

Comment:
perhaps consider logging if the metric already existed - even if just for now, so we don't mute out this error